### PR TITLE
*: get nil in the LFU's onEvict

### LINKED ISSUE #45946: panic: interface conversion: interface {} is nil, not *statistics.Table
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 3. What did you see instead (Required)
```
panic: interface conversion: interface {} is nil, not *statistics.Table
goroutine 234846 [running]:
[github.com/pingcap/tidb/statistics/handle/cache/internal/lfu.(*LFU).onEvict](http://github.com/pingcap/tidb/statistics/handle/cache/internal/lfu.(*LFU).onEvict)(0xc0713f33f8, 0xc01c5e1680)
        statistics/handle/cache/internal/lfu/lfu_cache.go:123 +0x1be
[github.com/dgraph-io/ristretto.NewCache.func2(0xc01c5e1680)](http://github.com/dgraph-io/ristretto.NewCache.func2(0xc01c5e1680))
        external/com_github_dgraph_io_ristretto/cache.go:188 +0x31
[github.com/dgraph-io/ristretto.(*Cache).processItems.func2(0xc01c5e1680)](http://github.com/dgraph-io/ristretto.(*Cache).processItems.func2(0xc01c5e1680))
        external/com_github_dgraph_io_ristretto/cache.go:447 +0xb7
[github.com/dgraph-io/ristretto.(*Cache).processItems(0xc00619c000)](http://github.com/dgraph-io/ristretto.(*Cache).processItems(0xc00619c000))
        external/com_github_dgraph_io_ristretto/cache.go:479 +0x3ae
created by [github.com/dgraph-io/ristretto.NewCache](http://github.com/dgraph-io/ristretto.NewCache) in goroutine 209915
        external/com_github_dgraph_io_ristretto/cache.go:207 +0x645
```

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #45946

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: statistics/handle/cache/internal/lfu/lfu_cache.go [hunk 1]
```diff
@@ -117,6 +124,11 @@ func DropEvicted(item statistics.TableCacheItem) {
 }
 
 func (s *LFU) onEvict(item *ristretto.Item) {
+	if item.Value == nil {
+		// Sometimes the same key may be passed to the "onevict" function twice,
```
Comment (winoros) [inline statistics/handle/cache/internal/lfu/lfu_cache.go:128]:
```suggestion
		// Sometimes the same key may be passed to the "onEvict" function twice,
```

File: statistics/handle/cache/internal/lfu/lfu_cache.go [hunk 2]
```diff
@@ -77,6 +79,11 @@ func (s *LFU) Get(tid int64, _ bool) (*statistics.Table, bool) {
 
 // Put implements statsCacheInner
 func (s *LFU) Put(tblID int64, tbl *statistics.Table) bool {
+	if tbl == nil {
+		logutil.BgLogger().Fatal("table is nil", zap.Int64("tableID", tblID))
+	} else {
+		logutil.BgLogger().Info("table", zap.Int64("tableID", tblID), zap.Stack("stack"))
```
Comment (qw4990) [inline statistics/handle/cache/internal/lfu/lfu_cache.go:85]: This log may cause some performance issues, better to remove this log.

Comment (hawkingrei) [reply statistics/handle/cache/internal/lfu/lfu_cache.go]: removed.

**Review by qw4990 — APPROVED** [review]

**Review by winoros — APPROVED** [review]

**Pull request merged**